File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannel.vb
'*******************************************************************************************************
'  IChannel.vb - Channel interface - this is the root interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Imports TVA.Parsing

''' <summary>This interface represents a protocol independent representation of any data type.</summary>
<CLSCompliant(False)> _
Public Interface IChannel

    Inherits IBinaryDataProvider

    ReadOnly Property DerivedType() As Type

    ReadOnly Property This() As IChannel

    ' At its most basic level - all data represented by the protocols can either be "parsed" or "generated"
    ' hence the following methods common to all elements

    Sub ParseBinaryImage(ByVal state As IChannelParsingState, ByVal binaryImage As Byte(), ByVal startIndex As Int32)

    Shadows ReadOnly Property BinaryLength() As UInt16

    ReadOnly Property Attributes() As Dictionary(Of String, String)

    Property Tag() As Object

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannel.cs
//*******************************************************************************************************
//  IChannel.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - J. Ritchie Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System.Collections.Generic;
using PCS.Parsing;

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of any data type that can
    /// be parsed or generated.<br/>
    /// This is the base interface implemented by all parsing/generating classes in the phasor
    /// protocols library; it is the root of the parsing/generating interface hierarchy.
    /// </summary>
    public interface IChannel : ISupportBinaryImage
    {
        /// <summary>
        /// Gets a <see cref="Dictionary{TKey,TValue}"/> of string based property names and values for this <see cref="IChannel"/> object.
        /// </summary>
        Dictionary<string, string> Attributes { get; }

        /// <summary>
        /// Gets or sets the parsing state for this <see cref="IChannel"/> object.
        /// </summary>
        IChannelParsingState State { get; set; }

        /// <summary>
        /// Gets or sets a user definable reference to an object associated with this <see cref="IChannel"/> object.
        /// </summary>
        /// <remarks>
        /// Classes implementing <see cref="IChannel"/> should only track this value.
        /// </remarks>
        object Tag { get; set; }
    }
}